File: service-common/src/main/java/me/reidj/bridgebuilders/packages/SaveUserPackage.java
package me.reidj.bridgebuilders.packages;

import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.RequiredArgsConstructor;
import me.reidj.bridgebuilders.user.Stat;
import ru.cristalix.core.network.CorePackage;

import java.util.UUID;

@Data
@EqualsAndHashCode(callSuper = false)
@RequiredArgsConstructor
public class SaveUserPackage extends CorePackage {

    // request
    private final UUID user;
    private final Stat userInfo;

    // no response

}

File: service-common/src/main/kotlin/me/reidj/bridgebuilders/protocol/SaveUserPackage.kt
package me.reidj.bridgebuilders.protocol

import me.reidj.bridgebuilders.data.Stat
import ru.cristalix.core.network.CorePackage
import java.util.*

/**
 * @project : BridgeBuilders
 * @author : Рейдж
 **/
data class SaveUserPackage(
    val uuid: UUID,
    val stat: Stat
): CorePackage()
